lib/rspec/core/source.rb

```diff
@@ -0,0 +1,71 @@
+RSpec::Support.require_rspec_core 'source/node'
+RSpec::Support.require_rspec_core 'source/token'
+
+module RSpec
+  module Core
+    # @private
+    # Represents a Ruby source file and provides access to AST and tokens.
+    class Source
+      attr_reader :source, :path
+
+      def self.from_file(path)
+        source = File.read(path)
+        new(source, path)
+      end
+
+      def initialize(source_string, path=nil)
+        @source = source_string
+        @path = path ? File.expand_path(path) : '(string)'
+      end
+
+      def lines
+        @lines ||= source.split("\n")
+      end
+
+      def ast
+        @ast ||= begin
+          require 'ripper'
```

Comment:
What is the benefit of requiring ripper here instead of the top of the file?  I thought this file is only loaded if it's being used to extract a snippet, and that can't be done w/o ripper, right?  Seems better to declare ripper as a dependency at the top of the file and not re-require in this method and in `tokens`.